nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement the correct_name method.'
```

Comment:
- [ OPTIONAL ] Greetings! 🌟 Your work on implementing a method that detects errors when a class or method hasn't been implemented is truly commendable. 👏 The only improvement I would recommend is not only raising the error but also providing information about which specific class and method triggered the error. 🚀 This additional detail would make debugging and troubleshooting a breeze! 🛠️

```suggestion
    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```